src/session.rs

```diff
@@ -798,57 +885,11 @@ where
             self.flush();
         }
 
-        let mut finished = false;
-        for _ in 0..64 {
-            if !self.state.is_normal() {
-                break;
-            }
-            match self.socket.poll() {
-                Ok(Async::Ready(Some(sub_stream))) => self.handle_sub_stream(sub_stream),
-                Ok(Async::Ready(None)) => {
-                    finished = true;
-                    self.state = SessionState::RemoteClose;
-                    break;
-                }
-                Ok(Async::NotReady) => {
-                    finished = true;
-                    break;
-                }
-                Err(err) => {
-                    finished = true;
-                    debug!("session poll error: {:?}", err);
-                    self.write_buf.clear();
-                    self.high_write_buf.clear();
-                    if !self.keep_buffer {
-                        self.read_buf.clear()
-                    }
-
-                    match err.kind() {
-                        ErrorKind::BrokenPipe
-                        | ErrorKind::ConnectionAborted
-                        | ErrorKind::ConnectionReset
-                        | ErrorKind::NotConnected
-                        | ErrorKind::UnexpectedEof => self.state = SessionState::RemoteClose,
-                        _ => {
-                            warn!("MuxerError: {:?}", err);
-                            self.event_output(SessionEvent::MuxerError {
-                                id: self.context.id,
-                                error: err.into(),
-                            });
-                            self.state = SessionState::Abnormal;
-                        }
-                    }
-
-                    break;
-                }
-            }
-        }
-        if !finished {
-            self.set_delay();
-        }
+        self.poll_inner_socket();
 
         self.recv_substreams();
 
+        self.quick_recv_service();
```

Comment:
Should be the same logic as `Service::user_task_poll()`, otherwise it's not high priority.